### PR TITLE
Savestates: Asynchronous compression

### PR DESCRIPTION
**Pull request by elad335**
Following #14822, makes the creation of savestates faster.

### CODE REVIEW

File: Utilities/JIT.cpp [hunk 1]
```diff
@@ -10,6 +10,9 @@
 #include "util/v128.hpp"
 #include "util/simd.hpp"
 #include "Crypto/unzip.h"
+
+#include "Emu/savestate_utils.hpp"
```
Comment (Megamouse) [inline Utilities/JIT.cpp:14]: This makes no sense

File: rpcs3/Crypto/unzip.cpp [hunk 2]
```diff
@@ -3,6 +3,8 @@
 
 #include <zlib.h>
 
+#include "Emu/savestate_utils.hpp"
```
Comment (Megamouse) [inline rpcs3/Crypto/unzip.cpp:6]: If you need to include this here, something is really wrong with your software design

Comment (elad335) [reply rpcs3/Crypto/unzip.cpp:6]: More like the naming of the file.

File: rpcs3/Emu/Memory/vm.cpp [hunk 3]
```diff
@@ -1633,12 +1633,27 @@ namespace vm
 
 			if (ar.is_writing())
 			{
-				for (usz i = 0; i < process_size; i += 128)
+				for (usz i = 0; i < process_size; i += 128 * 2)
 				{
+					const u64 sample64_1 = read_from_ptr<u64>(ptr, i);
+					const u64 sample64_2 = read_from_ptr<u64>(ptr, i + 128);
+
+					// Speed up testing in scenerations where it is likely non-zero data
```
Comment (Megamouse) [inline rpcs3/Emu/Memory/vm.cpp:1641]: What is a scenaration?

**Comment by Megamouse** [comment]
Before you optimize things, you should fix your regression that blocks the entire project

**Comment by Ordinary205** [comment]
I can confirm that this fixes an issue after installing PS3 firmware, it also boots the game properly after firmware install.
Fixed
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/13489444/RPCS3.log.gz)

**Pull request merged**